sklearn/preprocessing/data.py

```diff
@@ -671,15 +711,33 @@ class MaxAbsScaler(BaseEstimator, TransformerMixin):
     def __init__(self, copy=True):
         self.copy = copy
 
+    def _reset(self):
+        """Reset internal data-dependent state of the scaler, if necessary.
+
+        __init__ parameters are not touched.
+        """
+
+        # Checking one attribute is enough, becase they are all set together
+        # in partial_fit
+        if hasattr(self, 'scale_'):
+            del self.scale_
+            del self.n_samples_seen_
+            del self.max_abs_
+
     def fit(self, X, y=None):
         """Compute the maximum absolute value to be used for later scaling.
 
+        it always resets the object's internal state first.
```

Comment:
I would not put this here. This is a scikit-learn convention. Documenting it in some estimators and not in others seems confusing to me.